.gitignore

```diff
@@ -29,3 +29,6 @@ coverage.txt
 *.out
 test/e2e/coexistence/
 test/e2e/testresults/
+
+# Extra go module files
+go.work.sum
```

Comment:
I still don't understand what this file does, but I think ignoring it is a good idea.
If it turns out over time that we should not, we can fix it in another PR.

Reply:
So, this is behaving very weird on my machine.
I check out the PR code and then the `go.work.sum` file changes by itself and is not being ignored.
We can discuss in person to compare behaviour.

Reply:
I will remove this change from this PR. I didn't mean to add it to this PR was trying it our myself as well but forgot to remove it.